src/Widgets/DLNAPage.vala

```diff
@@ -101,4 +56,106 @@ public class Sharing.Widgets.DLNAPage : Granite.SimpleSettingsPage {
+
+        private Gtk.Label folder_name;
+
+        public MediaEntry (string media_type, string label, Backend.RygelConfigFile config_file) {
+            Object (
+                config_file: config_file,
+                media_type: media_type,
+                label: label
+            );
+        }
+
+        construct {
+            var folder_dir = replace_xdg_folders (config_file.get_media_type_folder (media_type));
+
+            var check = new Gtk.CheckButton () {
+                active = config_file.get_media_type_enabled (media_type),
+                valign = CENTER
+            };
+
+            var icon_name = "";
+            switch (media_type) {
+                case "music":
+                    icon_name = "audio-x-generic";
+                    break;
+                case "pictures":
+                    icon_name = "image-x-generic";
+                    break;
+                case "videos":
+                    icon_name = "video-x-generic";
+                    break;
+            }
+
+            var image = new Gtk.Image.from_icon_name (icon_name, DND) {
+                pixel_size = 32
+            };
+
+            var header = new Granite.HeaderLabel (label);
+
+            folder_name = new Gtk.Label ("") {
+                halign = START,
+                hexpand = true
+            };
+
+            var arrow = new Gtk.Image.from_icon_name ("view-more-horizontal-symbolic", BUTTON);
+
+            var location_button_box = new Gtk.Box (HORIZONTAL, 3);
+            location_button_box.add (folder_name);
+            location_button_box.add (arrow);
+
+            var location_button = new Gtk.Button () {
+                child = location_button_box
+            };
+
+            var location_dialog = new Gtk.FileChooserNative (
+                _("Select Screenshots Folder…"),
+                ((Gtk.Application) Application.get_default ()).active_window,
+                Gtk.FileChooserAction.SELECT_FOLDER,
+                _("Select"),
+                null
+            );
+            location_dialog.set_current_folder (folder_dir);
+
+            column_spacing = 12;
+            attach (check, 0, 0, 1, 2);
+            attach (image, 1, 0, 1, 2);
+            attach (header, 2, 0);
+            attach (location_button, 2, 1);
+
+            check.bind_property ("active", image, "sensitive");
+
+            check.toggled.connect (() => {
+                config_file.set_media_type_enabled (media_type, check.active);
+                config_file.save ();
+            });
+
+            location_button.clicked.connect (() => {
+                location_dialog.run ();
+            });
+
+            folder_name.label = folder_dir;
+            location_dialog.response.connect ((response) => {
+                if (response == Gtk.ResponseType.ACCEPT) {
+                    var file = location_dialog.get_file ();
+                    folder_name.label = file.get_path ();
```

Comment:
Missing actually setting the config here. Something like:

```
rygel_config_file.set_media_type_folder (media_type_id, entry_file_chooser.get_file ().get_path ());
rygel_config_file.save ();
```

Reply:
Woops, thank you!